core/src/commonJvmMain/kotlin/com/ustadmobile/core/torrent/UstadTorrentManagerImpl.kt

```diff
@@ -0,0 +1,121 @@
+package com.ustadmobile.core.torrent
+
+import com.turn.ttorrent.client.*
+import com.turn.ttorrent.client.storage.FairPieceStorageFactory
+import com.turn.ttorrent.tracker.TrackedTorrent
+import com.turn.ttorrent.tracker.Tracker
+import com.ustadmobile.core.account.Endpoint
+import com.ustadmobile.core.io.ext.UstadFileCollectionStorage.createFileCollectionStorage
+import com.ustadmobile.core.util.DiTag
+import kotlinx.coroutines.*
+import org.kodein.di.DI
+import org.kodein.di.direct
+import org.kodein.di.on
+import org.kodein.di.instance
+import org.kodein.di.instanceOrNull
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import java.io.File
+
+
+interface CommunicationManagerListener {
+
+    fun onCommunicationManagerChanged(
+            communicationManager: UstadCommunicationManager?,
+            meteredNetwork: Boolean
+    )
+}
+
+class UstadTorrentManagerImpl(
+        val endpoint: Endpoint,
+        override val di: DI
+) : UstadTorrentManager, CommunicationManagerListener {
+
+    private val torrentDir: File by di.on(endpoint).instance(tag = DiTag.TAG_TORRENT_DIR)
+
+    private val containerDir by di.on(endpoint).instance<File>(tag = DiTag.TAG_DEFAULT_CONTAINER_DIR)
+
+    private var containerInfoHashMap = mutableMapOf<Long, String>()
```

Comment:
These multiple maps should be replaced with a single list that uses a data class with the info needed.

Reply:
These maps should be val instead of var.